src/core/public/application/scoped_history.mock.ts

```diff
@@ -51,7 +52,7 @@ const createMock = ({
     },
   };
 
-  return mock;
+  return mock as ScopedHistoryMock;
```

Comment:
So, I thought about creating a `IScopedHistory` interface, however the concrete `ScopedHistory` type is widely used in the codebase, so it was causing a lot of changes, which is why I decided to force cast to `jest.Mocked<ScopedHistory>` instead.

I think this is fine because:
- this is only used for testing
- we are still ensuring that the mock implements all public methods/properties of `ScopedHistory` with the `const mock: jest.Mocked<Pick<ScopedHistory, keyof ScopedHistory>>` declaration, so any changes in `ScopedHistory` public API would cause a TS error until the mock construction is adapted.
- only private methods and properties are actually missing from the returned mock instance.

Reply:
Could you add a comment on why the casting is necessary?